tests/integration/test_migrate.py

```diff
@@ -3,11 +3,53 @@
 Licensed under your choice of the Redis Source Available License 2.0 (RSALv2)
 or the Server Side Public License v1 (SSPLv1).
 """
+import logging
+
+import redis
 
 from _pytest.python_api import raises
 from redis import ResponseError
 
-from .sandbox import RawConnection, RedisRaft
+from .sandbox import RawConnection, RedisRaft, RedisRaftTimeout
+
+
+def migrate_slots(cluster, slots):
+    cursor = 0
+    max_retries = 100
+
+    while True:
+        leader = cluster.leader_node()
+
+        try:
+            reply = leader.execute('raft.scan', str(cursor), slots)
+            cursor = int(reply[0])
+            keys = reply[1]
+
+            if len(keys) != 0:
+                key_names = []
+                for key in keys:
+                    key_names.append(key[0].decode('utf-8'))
+
+                assert leader.execute('migrate', '', '', '', '', '', 'keys',
+                                      *key_names) == b'OK'
+
+            # If cursor is zero, we've moved all the keys
+            if cursor == 0:
+                break
+
+        except (redis.ResponseError, redis.TimeoutError) as e:
+            logging.info('raft.scan command failed with error: ' + str(e))
+
+            max_retries -= 1
+            if max_retries == 0:
+                raise RedisRaftTimeout("migrate command was not successful")
+
+            # In case of error, we just update the leader node and start from
+            # beginning. If leader has changed, our current 'cursor' value
+            # is not valid on the new leader.
```

Comment:
I realized one thing and I think we didn't discuss this before. On leader change, admin should reset `cursor` value. (it should start from beginning). 

`cursor` iteration should be valid on the same node only IIUC. Not sure if ngc was aware of this as well. 

@sjpotter @fadidahanna wdyt?